pkg/controller/schedulingmanager/controller.go

```diff
@@ -17,63 +17,78 @@ limitations under the License.
 package schedulingmanager
 
 import (
-	"sync"
-
 	"github.com/golang/glog"
 	"github.com/pkg/errors"
 
+	"k8s.io/apimachinery/pkg/util/runtime"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+
 	"github.com/kubernetes-sigs/federation-v2/pkg/apis/core/typeconfig"
 	corev1a1 "github.com/kubernetes-sigs/federation-v2/pkg/apis/core/v1alpha1"
 	"github.com/kubernetes-sigs/federation-v2/pkg/controller/schedulingpreference"
 	"github.com/kubernetes-sigs/federation-v2/pkg/controller/util"
 	"github.com/kubernetes-sigs/federation-v2/pkg/schedulingtypes"
-
-	"k8s.io/apimachinery/pkg/util/runtime"
-	"k8s.io/apimachinery/pkg/util/sets"
-	restclient "k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/cache"
 )
 
-type SchedulerController struct {
-	sync.RWMutex
+type SchedulingManager struct {
 	// Store for the FederatedTypeConfig objects
 	store cache.Store
 	// Informer for the FederatedTypeConfig objects
 	controller cache.Controller
 
 	worker util.ReconcileWorker
 
-	scheduler map[string]schedulingtypes.Scheduler
+	//schedulers map[string]*SchedulerWrapper
+	schedulers *util.SafeMap
 
 	config *util.ControllerConfig
+}
 
-	runningPlugins sets.String
-	// mapping qualifiedname to template kind for managing plugins in scheduler
-	federatedKindMap map[string]string
+type SchedulerWrapper struct {
```

Comment:
(No action necessary) I'm not sure I understand the point of SchedulerWrapper.  What is the advantage to including these fields here vs on `schedulingtypes.Scheduler` (along with the necessary methods to interact with them)?